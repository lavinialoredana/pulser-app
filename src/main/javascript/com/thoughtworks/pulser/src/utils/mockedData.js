export const mockedDataArray = [
  {
    id: "63c8121120b26e6a06d92c19",
    face: "AWESOME",
    body: "having a great time!",
  },
  {
    id: "63c8122320b26e6a06d92c1a",
    face: "HAPPY",
    body: "this is really nice",
  },
  {
    id: "63c8125220b26e6a06d92c1b",
    face: "SAD",
    body: "well... not so good",
  },
  {
    id: "63c8128320b26e6a06d92c1c",
    face: "AWFUL",
    body: "not good at all, not gonna lie",
  },
];
