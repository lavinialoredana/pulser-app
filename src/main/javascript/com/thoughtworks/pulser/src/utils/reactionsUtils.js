import awesomeFace from "../assets/images/awesome_face.png"
import happyFace from "../assets/images/happy_face.png";
import sadFace from "../assets/images/sad_face.png";
import angryFace from "../assets/images/angry_face.png";

export const reactionsArray = [
  {
    key: "awesomeFace_key",
    reactionName: "Awesome",
    reactionImage: awesomeFace,
    reactionAlt: "awesome_face",
  },
  {
    key: "happyFace_key",
    reactionName: "Happy",
    reactionImage: happyFace,
    reactionAlt: "happy_face",
  },
  {
    key: "sadFace_key",
    reactionName: "Sad",
    reactionImage: sadFace,
    reactionAlt: "sad_face",
  },
  {
    key: "angryFace_key",
    reactionName: "Angry",
    reactionImage: angryFace,
    reactionAlt: "angry_face",
  },
];
