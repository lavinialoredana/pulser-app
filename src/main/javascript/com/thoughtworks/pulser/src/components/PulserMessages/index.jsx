import React from "react";
import { reactionsArray } from "../../utils/reactionsUtils";
import PulserMessage from "../PulserMessage/index";

const PulserMessages = ({ feedData }) => {
  return feedData.map((message) => {
    for (let i = 0; i < reactionsArray.length; i++) {
      if (message.face === reactionsArray[i].key) {
        return (
          <div className="Pulser-messages-container" key={message.id}>
            <PulserMessage
              reactionTitle={reactionsArray[i].reactionName}
              reactionFace={reactionsArray[i].reactionImage}
              reactionFaceAlt={reactionsArray[i].reactionAlt}
              feedbackMessage={message.body}
            />
          </div>
        );
      }
    }
    return <p> We are trying to display your data!</p>;
  });
};

export default PulserMessages;
