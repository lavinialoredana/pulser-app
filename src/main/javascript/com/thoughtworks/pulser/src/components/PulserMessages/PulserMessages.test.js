import { cleanup, render, screen } from "@testing-library/react";
import PulserMessages from "./index.jsx";
import { mockedDataArray } from "../../utils/mockedData";

describe("PulserMessages component", () => {
  afterEach(() => {
    cleanup();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <PulserMessages feedData={mockedDataArray} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should contain Awesome title and awesome_face image", () => {
    render(<PulserMessages feedData={mockedDataArray} />);

    const titleText = screen.getByText("Awesome");
    const awesomeImage = screen.getByAltText("awesome_face");

    expect(titleText).toBeInTheDocument();
    expect(awesomeImage.src).toContain("awesome_face.png");
  });

  it("should match data length with feed length", () => {
    render(<PulserMessages feedData={mockedDataArray} />);

    const messageElement = screen.getAllByTestId("reaction-message");

    expect(messageElement.length).toEqual(mockedDataArray.length);
  });
});
