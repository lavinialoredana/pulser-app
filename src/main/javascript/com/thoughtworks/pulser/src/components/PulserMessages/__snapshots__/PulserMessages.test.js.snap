// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PulserMessages component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Pulser-messages-container"
  >
    <div
      class="Pulser-message-container"
    >
      <div
        class="Title-image-reaction-container"
      >
        <div
          class="Reaction-title"
          data-testid="reaction-title"
        >
          <h3>
             Awesome
          </h3>
        </div>
        <div>
          <img
            alt="awesome_face"
            class="Reaction-image"
            src="awesome_face.png"
          />
        </div>
      </div>
      <div
        class="Reaction-message"
        data-testid="reaction-message"
      >
        <p>
          having a great time!
        </p>
      </div>
    </div>
  </div>
  <div
    class="Pulser-messages-container"
  >
    <div
      class="Pulser-message-container"
    >
      <div
        class="Title-image-reaction-container"
      >
        <div
          class="Reaction-title"
          data-testid="reaction-title"
        >
          <h3>
             Happy
          </h3>
        </div>
        <div>
          <img
            alt="happy_face"
            class="Reaction-image"
            src="happy_face.png"
          />
        </div>
      </div>
      <div
        class="Reaction-message"
        data-testid="reaction-message"
      >
        <p>
          this is really nice
        </p>
      </div>
    </div>
  </div>
  <div
    class="Pulser-messages-container"
  >
    <div
      class="Pulser-message-container"
    >
      <div
        class="Title-image-reaction-container"
      >
        <div
          class="Reaction-title"
          data-testid="reaction-title"
        >
          <h3>
             Sad
          </h3>
        </div>
        <div>
          <img
            alt="sad_face"
            class="Reaction-image"
            src="sad_face.png"
          />
        </div>
      </div>
      <div
        class="Reaction-message"
        data-testid="reaction-message"
      >
        <p>
          well... not so good
        </p>
      </div>
    </div>
  </div>
  <div
    class="Pulser-messages-container"
  >
    <div
      class="Pulser-message-container"
    >
      <div
        class="Title-image-reaction-container"
      >
        <div
          class="Reaction-title"
          data-testid="reaction-title"
        >
          <h3>
             Angry
          </h3>
        </div>
        <div>
          <img
            alt="angry_face"
            class="Reaction-image"
            src="angry_face.png"
          />
        </div>
      </div>
      <div
        class="Reaction-message"
        data-testid="reaction-message"
      >
        <p>
          not good at all, not gonna lie
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
